### PR TITLE
replacing malloc+memset with calloc

### PR DESCRIPTION
**Pull request by UserXGnu**
I know that speed is not an issue here and changing something so tiny like that wouldn't make that difference in the performance at all, but the calloc does the same of malloc + memset with better performance and with less function calls which could help keeping the code cleaner. 

So, here it goes the shy PR.

### CODE REVIEW

File: hdump.c [hunk 1]
```diff
@@ -116,12 +114,15 @@ int main(int argc, char *argv[])
 			/* imprime os bytes separados por espaço */
 			printf("%02x%*c", (unsigned int) *(buff+i), (i+1 == cols/2) ? 2 : 1, ' ');
 
-			/* define o fim do array ascii (sera usado como string) */
-			*(ascii+bread) = '\0';
-
-			/* imprime os caracteres ascii */
-			if (i == bread-1)
```
Comment (UserXGnu) [inline hdump.c]: I replaced the ascii assignement to define its end inside the if condition once that this assignment might happen only after every byte gets printed, this way we can avoid unnecessary assignment into ascii every time the loop restart.

Comment (UserXGnu) [inline hdump.c]: I moved the ascii assignement to define its end to inside the if condition once that this assignment might happen only after every bread gets printed, this way we can avoid unnecessary assignment into ascii every time the loop restart.

**Comment by UserXGnu** [comment]
In order to supply some kind of fix to the "large file" issue, I added this last commit which partially fix it allowing the user to execute hdump with file's even larger than 4GB, however, this fix only work on unix-like operating systems, for Windows the change doesn't cause any side effect.

**Comment by merces** [comment]
Amazing! Thanks!

**Pull request merged**